includes/FormFactory/ManageWikiFormFactoryBuilder.php

```diff
@@ -631,10 +631,45 @@ private static function buildDescriptorPermissions(
 
 		$userGroupManager = MediaWikiServices::getInstance()->getUserGroupManager();
 
+		$assignedPermissions = $groupData['permissions'] ?? [];
+
+		$disallowed = isset( $config->get( 'ManageWikiPermissionsDisallowedRights' )[$group] ) ?
+			array_merge(
+				$config->get( 'ManageWikiPermissionsDisallowedRights' )[$group],
+				$config->get( 'ManageWikiPermissionsDisallowedRights' )['any']
+			) :
+			$config->get( 'ManageWikiPermissionsDisallowedRights' )['any'];
+
+		$allPermissions = MediaWikiServices::getInstance()->getPermissionManager()->getAllPermissions();
+
+		// Start with all allowed permissions
+		$allPermissions = array_diff( $allPermissions, $disallowed );
+
+		/**
+		 * Include any assigned permissions that aren’t in
+		 * current available permissions and aren’t disallowed.
+		 * This may include permissions for extensions that have
+		 * since been disabled. We do this so that the permissions
+		 * can be removed from them, otherwise the groups that use
+		 * these permissions become undeletable.
+		 */
+		$extraAssigned = array_filter(
+			$assignedPermissions,
+			static fn ( string $perm ): bool => !in_array( $perm, $allPermissions, true ) &&
+				!in_array( $perm, $disallowed, true )
```

Comment:
`$disallowed` is already removed from `$allPermissions` from the above `array_diff()`, therefore:
```suggestion
			static fn ( string $perm ): bool => !in_array( $perm, $allPermissions, true )
```

Reply:
But not from here, since this checks for those assigned permissions that arent in all permissions if we don't check for disallowed here, it will readd disallowed back to all since they aren't in the array.

Reply:
Wait do we want disallowed permissions to be removalable or not at all? If we do we can remove that check and assigned but disallowed would be removalable but not addable. I'm not sure if we want to do that though...

Reply:
Can continue this on the parent PR